x-pack/plugins/canvas/server/lib/normalize_type.ts

```diff
@@ -24,10 +24,11 @@ export function normalizeType(type) {
     null: ['null'],
   };
 
-  const normalizedType = Object.keys(normalTypes).find(t => normalTypes[t].includes(type));
-
-  if (normalizedType) {
-    return normalizedType;
+  for (const [normalizedType, inputTypes] of Object.entries(normalTypes)) {
```

Comment:
This is tough to follow.  Perhaps TS enums would help?

https://www.typescriptlang.org/docs/handbook/enums.html

Reply:
I'm curious what you find hard to follow, and would be interested in learning how an enum here would help.